From fa27358c48772ddea52f10fc85d14dbad0ec20a8 Mon Sep 17 00:00:00 2001
From: Christian Ruppert <idl0r@gentoo.org>
Date: Fri, 19 Nov 2010 20:53:37 +0100
Subject: [PATCH 2/2] Fix cf-key --output-file/-f option

CFPRIVKEYFILE and CFPUBKEYFILE has been overriden by CheckWorkingDirectories().
We now check if both variables are empty (which is the default, without
		--output-file/-f).
---
 src/generic_agent.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/generic_agent.c b/src/generic_agent.c
index 8287bbc..483b20f 100644
--- a/src/generic_agent.c
+++ b/src/generic_agent.c
@@ -1106,8 +1106,10 @@ if (cfstat(CFWORKDIR,&statbuf) != -1)
 snprintf(vbuff,CF_BUFSIZE,"%s%cstate%c.",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
 MakeParentDirectory(vbuff,false);

-snprintf(CFPRIVKEYFILE,CF_BUFSIZE,"%s%cppkeys%clocalhost.priv",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
-snprintf(CFPUBKEYFILE,CF_BUFSIZE,"%s%cppkeys%clocalhost.pub",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
+if(strlen(CFPRIVKEYFILE) == 0 && strlen(CFPUBKEYFILE) == 0) {
+	snprintf(CFPRIVKEYFILE,CF_BUFSIZE,"%s%cppkeys%clocalhost.priv",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
+	snprintf(CFPUBKEYFILE,CF_BUFSIZE,"%s%cppkeys%clocalhost.pub",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
+}

 CfOut(cf_verbose,"","Checking integrity of the state database\n");
 snprintf(vbuff,CF_BUFSIZE,"%s%cstate",CFWORKDIR,FILE_SEPARATOR,FILE_SEPARATOR);
--
1.7.2.2

